spark/v3.5/spark-extensions/src/main/scala/org/apache/spark/sql/catalyst/analysis/ViewCheck.scala

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.spark.sql.catalyst.analysis
+
+import org.apache.iceberg.spark.Spark3Util
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
+import org.apache.spark.sql.catalyst.plans.logical.views.CreateIcebergView
+import org.apache.spark.sql.connector.catalog.ViewCatalog
+import org.apache.spark.sql.execution.command.ViewHelper
+
+object ViewCheck extends (LogicalPlan => Unit) {
+
+  override def apply(plan: LogicalPlan): Unit = {
+    plan foreach {
+      case CreateIcebergView(ResolvedIdentifier(_: ViewCatalog, ident), _, _, _, _, query, _, _) =>
+        val identifier = Spark3Util.toV1TableIdentifier(ident)
+        ViewHelper.verifyTemporaryObjectsNotExists(false, identifier, query, Seq.empty)
```

Comment:
ideally we'd have these checks be part of `CreateV2ViewExec` but by the time we reach `CreateV2ViewExec` the underlying `query' doesn't contain the info needed to detect these cases here.

Reply:
If the identifier is a `ResolvedIdentifier`, don't we already know that a temporary view does not exist? That's checked when resolving views. Does the `UnresolvedIdentifier` has `allowTemp` set to false?

Reply:
what the check does is to make sure that a view doesn't reference a temporary/global view. We particularly test for these 2 cases in `createViewReferencingGlobalTempView()` / `createViewReferencingTempView()`